src/http/client.c

```diff
@@ -22,43 +23,74 @@
 
 enum {
 	CONN_TIMEOUT = 30000,
-	RECV_TIMEOUT = 30000,
+	RECV_TIMEOUT = 60000,
+	IDLE_TIMEOUT = 900000,
 	BUFSIZE_MAX  = 524288,
+	CONN_BSIZE   = 256,
 };
 
 struct http_cli {
+	struct list reql;
+	struct hash *ht_conn;
 	struct dnsc *dnsc;
 	struct tls *tls;
 };
 
+struct conn;
+
 struct http_req {
 	struct sa srvv[16];
-	struct tmr tmr;
+	struct le le;
 	struct http_req **reqp;
+	struct http_cli *cli;
 	struct dns_query *dq;
-	struct tls_conn *sc;
-	struct tcp_conn *tc;
+	struct conn *conn;
 	struct mbuf *mbreq;
 	struct mbuf *mb;
-	struct tls *tls;
 	char *host;
 	http_resp_h *resph;
 	http_data_h *datah;
 	void *arg;
+	size_t rx_bytes;
+	size_t rx_len;
 	unsigned srvc;
 	uint16_t port;
 	bool secure;
 	bool data;
 };
 
 
+struct conn {
+	struct tmr tmr;
+	struct sa addr;
+	struct le he;
+	struct http_req *req;
+	struct tls_conn *sc;
+	struct tcp_conn *tc;
+	uint64_t usec;
```

Comment:
when I first read this line I was thinking "micro-seconds"

when reading the code I found out it meant "usecount".

perhaps rename to "use_count" and use a shorter datatype?